src/cdk-experimental/listbox/listbox.ts

```diff
@@ -8,182 +8,146 @@
-import {SelectionChange, SelectionModel} from '@angular/cdk/collections';
-import {defer, merge, Observable, Subject} from 'rxjs';
-import {startWith, switchMap, takeUntil} from 'rxjs/operators';
+import {SelectionModel} from '@angular/cdk/collections';
+import {BehaviorSubject, defer, merge, Observable, Subject} from 'rxjs';
+import {filter, mapTo, startWith, switchMap, takeUntil} from 'rxjs/operators';
 import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
 import {Directionality} from '@angular/cdk/bidi';
 import {CDK_COMBOBOX, CdkCombobox} from '@angular/cdk-experimental/combobox';
 
-let nextId = 0;
-let listboxId = 0;
+/** The next id to use for the CdkOption directive. */
+let nextOptionId = 0;
 
-export const CDK_LISTBOX_VALUE_ACCESSOR = {
-  provide: NG_VALUE_ACCESSOR,
-  useExisting: forwardRef(() => CdkListbox),
-  multi: true,
-};
+/** The next id to use for the CdkListbox directive. */
+let nextListboxId = 0;
 
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     'role': 'option',
     'class': 'cdk-option',
-    '(click)': 'toggle()',
-    '(focus)': 'activate()',
-    '(blur)': 'deactivate()',
     '[id]': 'id',
-    '[attr.aria-selected]': 'selected || null',
-    '[attr.tabindex]': '_getTabIndex()',
-    '[attr.aria-disabled]': '_isInteractionDisabled()',
-    '[class.cdk-option-disabled]': '_isInteractionDisabled()',
-    '[class.cdk-option-active]': '_active',
-    '[class.cdk-option-selected]': 'selected',
+    '[attr.aria-selected]': 'isSelected() || null',
+    '[attr.tabindex]': '!listbox.useActiveDescendant && !disabled ? -1 : null',
+    '[attr.aria-disabled]': 'disabled',
+    '[class.cdk-option-disabled]': 'disabled',
+    '[class.cdk-option-active]': 'isActive()',
+    '[class.cdk-option-selected]': 'isSelected()',
+    '(click)': '_clicked.next()',
   },
 })
-export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightable {
-  private _selected: boolean = false;
-  private _disabled: boolean = false;
-  private _value: T;
-  _active: boolean = false;
+export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightable, OnDestroy {
+  /** The id of the option's host element. */
+  @Input() id = `cdk-option-${nextOptionId++}`;
 
-  /** The id of the option, set to a uniqueid if the user does not provide one. */
-  @Input() id = `cdk-option-${nextId++}`;
+  // TODO(mmalerba): What do we do if the user doesn't specify a value?
+  /** The value of this option. */
+  @Input('cdkOptionValue') value: T;
 
-  @Input()
-  get selected(): boolean {
-    return this._selected;
-  }
-  set selected(value: BooleanInput) {
-    if (!this._disabled) {
-      this._selected = coerceBooleanProperty(value);
-    }
-  }
+  /**
+   * The text used to locate this item during listbox typeahead. If not specified,
+   * the `textContent` of the item will be used.
+   */
+  @Input('cdkOptionTypeaheadLabel') typeaheadLabel: string;
```

Comment:
Did this exist already or is this a new addition? Either I don't remember or don't understand the context for this

Reply:
I added this at the same time I added it to the cdk menu. There was logic that was looking for mat-icon and excluding it from the typeahead text, but we decided that the cdk shouldn't have magic knowledge of material and opted to instead provide this input to allow users to supply their own function to get the typeahead label